python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -185,22 +185,13 @@ cpdef read_parquet(filepaths_or_buffers, columns=None, row_groups=None,
 
     update_struct_field_names(df, c_out_table.metadata.schema_info)
 
-    if df.empty and meta is not None:
-        cols_dtype_map = {}
-        for col in meta['columns']:
-            cols_dtype_map[col['name']] = col['numpy_type']
-
-        if not column_names:
-            column_names = [o['name'] for o in meta['columns']]
-            if not is_range_index and index_col in cols_dtype_map:
-                column_names.remove(index_col)
-
-        for col in column_names:
-            meta_dtype = cols_dtype_map.get(col, None)
-            df._data[col] = cudf.core.column.column_empty(
-                row_count=0,
-                dtype=cudf.dtype(meta_dtype)
-            )
```

Comment:
@galipremsagar do you know what case(s) this is meant to handle? AFAICT, we still pass tests after removing these lines (although CI may prove me wrong).

Reply:
Welp..

Reply:
I think this was a case special case to handle the round-tripping of an empty data frame to preserve types...Looking into how it is passing.

Reply:
This was primarily inplace to handle empty dataframes where the `pandas_type` & `numpy_type` differ, for ex:
```python
{'name': 'b', 'field_name': 'b', 'pandas_type': 'bool', 'numpy_type': 'boolean', 'metadata': None},
{'name': 'b', 'field_name': 'b', 'pandas_type': 'int8', 'numpy_type': 'bool', 'metadata': None},
{'name': 'c', 'field_name': 'c', 'pandas_type': 'unicode', 'numpy_type': 'object', 'metadata': None}
```

Okay, I did some debugging and seems like `c_out_table` is "newly" having all the correct types incase of an empty dataframe too. This seems like to be fixed at libcudf level, probably due to the CUIO refactor so I think this is safe to be removed now. Which PR fixed this I'm not completely sure, @devavret | @vuule do you know if any you made a fix related to this at libcudf level?

Reply:
I remember that @nvdbaranec made changes to how empty columns are created to fix struct related issues. Maybe that PR modified the related libcudf behavior here as well.